### ISSUE TITLE
Cannot connect to VPN

### ISSUE THREAD

**Issue by FlailAway** [mate-desktop/mate-desktop#618]
#### Expected behaviour
Click Network and select an Installed VPN

#### Actual behaviour
Password loop with the correct password showing. Clicking the dialog I wait for about 5-seconds and reappears.

#### Steps to reproduce the behaviour
Left-click network icon in panel.
Click VPN settings
select a VPN

#### MATE general version
See below

#### Package version
See below

#### Linux Distribution
Linux Mint-22+MATE

#### Link to bugreport of your Distribution (requirement)
I do not know what this means?

I cannot give you the MATE Version above as I am using Cinnamon because I cannot connect to our VPN with MATE.

This is a fresh install of the current Mint-22+MATE downloaded yesterday. I have tried on 3-different laptops and all the same. Two were upgraded using the MintUpgrade tool, but this computer is brand new and first install direct form .iso. Same issue.

I have installed Cinnamon desktop in this former MATE version and the VPN is working fine.  I guess I can reboot to MATE desktop and then get the versions, reboot to Cinnamon and come back and edit this post.

ETA: I have been using Mint 21.3+MATE for a year or more. If I use Timeshift to roll back the Mint-22+MATE to 21.3, the VPN works fine as it always has.

**Comment by FlailAway** [comment]
OK, thanks for the reply, but why does it only fail with MATE in Mint-22? It seems horribly like a MATE problem with Mint-22 as Mint-21.3 + MATE have been working fine for  year or more.

If something **only** happens with MATE in Mint-22, how can it not be something going on with a process in the MATE desktop? 

Could it be something to do with password management perhaps?

I don't see how one can wash their hands of this when it specifically behaves differently from Mint-21.3 (I am back to using) and Mint-22.

In Mint-22, I deleted the VPN from the network app and also deleted all passwords. I created a new Ethernet connection and a new VPN instance. Shut down, restarted and still it goes to the VPN Password dialog and loops there.

I will reinstate (Timeshift) Mint-22+MATE and get the details of the MATE version.

Back in a while with the MATE data.

**Comment by lukefromdc** [comment]
Could be a difference between GNOME's network-manager-applet (used in MATE) and the applet Cinnamon uses.

**Comment by FlailAway** [comment]
OK, finally resolved the issue. It is probably deeper than just MATE and maybe even deeper than a Mint-thing. 

When I set up this system I used the "no login at start" in the .iso installer.

When I created the VPN access it asked for a keyring password (none at login) and stupidly created a new "default login" keyring **instead of opening** the existing and empty "login" keyring. This left the "login" password empty but still set as "default." 

When the network tried  to connect to the VPN, it looked in the empty "login" keyring and declared an error that displayed the "password-required" prompt. BUT the dialog showed the password from "default password" keyring. Crazy stuff. The dialog knew to look in the "default login" keyring the network not so much, as it again checked the empty "login" keyring and around we went.

To resolve this, in "Passwords"  I set (right-click) the empty "logon" to "default" and then deleted the "default login" keyring. I deleted the VPN entry from the network settings and reinstalled the VPN. This time it used "login" and added the password to it. Finally all good. GNOME/Ubuntu/Mint devs. et-al should probably read this. {grin}
